### PR TITLE
Add support for filtering defaults based on firmware version

### PR DESCRIPTION
**Pull request by etracer65**
Ensures that when filters are enabled the defaults used are appropriate for the firmware version.

### CODE REVIEW

File: src/js/tabs/pid_tuning.js [hunk 1]
```diff
@@ -13,12 +13,28 @@ TABS.pid_tuning = {
...
+    } else if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
+        FILTER_DEFAULT.gyro_lowpass_hz = 150;
+        FILTER_DEFAULT.gyro_lowpass_type = 1;   // BIQUAD
+        FILTER_DEFAULT.gyro_lowpass2_hz = 0;
+        FILTER_DEFAULT.gyro_lowpass2_type = 1;  // BIQUAD
+        FILTER_DEFAULT.dterm_lowpass_hz = 150;
+        FILTER_DEFAULT.dterm_lowpass_type = 1;  // BIQUAD
+        FILTER_DEFAULT.dterm_lowpass2_hz = 150;
+        FILTER_DEFAULT.dterm_lowpass2_type = 1;  // BIQUAD
+    }
```
Comment (McGiverGim) [inline src/js/tabs/pid_tuning.js]: Not sure if this is the best approach. We have defaults defined in the fc.js file (I think remember) that are modified here. I think it's better to have them in only one place, two options:
- Change in the fc.js the defaults by a function that returns them based on the version.
- Move the defaults here. I think we only have PID and filter defaults if I'm not wrong in all the Configurator.
What do you think?

Comment (etracer65) [reply src/js/tabs/pid_tuning.js]: I'm going to be out until late tonight, but I'll look at adding a `getFilterDefaults()` function to fc.js and use that to return the values based on version. Then all the maintenance will be in fc.js.

File: src/js/tabs/pid_tuning.js [hunk 2]
```diff
@@ -13,12 +13,28 @@ TABS.pid_tuning = {
...
         GUI.active_tab = 'pid_tuning';
     }
 
+    // Update filtering defaults based on API version
+    var FILTER_DEFAULT = DEFAULT
+    if (semver.eq(CONFIG.apiVersion, "1.40.0")) {
+        FILTER_DEFAULT.dterm_lowpass2_hz = 200;
+    } else if (semver.gte(CONFIG.apiVersion, "1.41.0")) {
+        FILTER_DEFAULT.gyro_lowpass_hz = 150;
+        FILTER_DEFAULT.gyro_lowpass_type = 1;   // BIQUAD
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js]: Using a 'constant' for `BIQUAD` will make this look much cleaner.

File: src/js/tabs/pid_tuning.js [hunk 3]
```diff
@@ -13,12 +13,28 @@ TABS.pid_tuning = {
...
 TABS.pid_tuning.initialize = function (callback) {
+
     var self = this;
 
     if (GUI.active_tab !== 'pid_tuning') {
         GUI.active_tab = 'pid_tuning';
     }
 
+    // Update filtering defaults based on API version
+    var FILTER_DEFAULT = DEFAULT
```
Comment (mikeller) [inline src/js/tabs/pid_tuning.js]: Semicolon missing.

Comment (mikeller) [inline src/js/tabs/pid_tuning.js]: Semicolon missing.

**Comment by etracer65** [comment]
Revised per feedback and pushed an update.

**Comment by mikeller** [comment]
This works, but I am not sure if it works in the best possible way:

When starting out with defaults, then entering the PID Tuning tab, and saving, the `diff` shows these artefacts:

```
# master
set gyro_lowpass_hz = 0
set gyro_lowpass2_type = PT1

# profile
profile 0

set dterm_lowpass_hz = 0
```

I think `gyro_lowpass2_type = PT1` is definitely unwanted.

We have to decide if setting the lowpass to `0` is the lesser evil (this was already reported and 'fixed' once, #1333), or if we should just leave the lowpass 1 cutoff settings as they are (i.e. the default / user set value is retained even when they are off).

Another thing to consider is if we want to add an extra warning text that shows when a user disables both lowpass 1 and dynamic lowpass, to stop users from getting dangerous flyaways.

**Comment by McGiverGim** [comment]
I think I did it for all the filters some versions ago. When disabled, I put to zero (or the first value of the select) all the parameters of the filter, to remark that is disabled. I never thought about the `diff` problem.

**Comment by mikeller** [comment]
@McGiverGim: I think in general and for other settings this is the right approach. But in this case, since we cannot use a default of 0 for these settings even though they are disabled by default, setting them to 0 in configurator will cause users to report this as a bug, like it happened in #1333.

**Comment by etracer65** [comment]
Yes, we need to fix this up and not mess with the settings. Actually setting `gyro_lowpass2_type = PT1` isn't that bad because it's off by default (and the reason it's being changed). It will get reset to the proper BIQUAD setting if reenabled in the Configurator with changes in this PR. But it's still an undesirable settings change that will show up in the `diff` for no reason. Also might hang people out if they reenable the filter in the CLI by just updating the cutoff but forgetting to reset the type.

Setting `gyro_lowpass_hz` and `dterm_lowpass_hz` to 0 is bad and will trigger the earlier problem if somebody happens to use the 10.4 configurator later.

If you want to point me to where this is being done I can include in this PR. Or is not appropriate we can do a separate PR to correct as it's not directly related to this PR about how the defaults are set.

**Comment by McGiverGim** [comment]
@etracer65 you will find it here:

https://github.com/betaflight/betaflight-configurator/blob/a36329c8606b4dd4d5ee4bb6ca57a72c95c8bee5/src/js/tabs/pid_tuning.js#L456

Similar for others filters.

**Comment by etracer65** [comment]
@McGiverGim Thanks.

I think I'll just avoid the resets entirely if API >= 1.41. They're ok for older firmware versions but cause problems in 4.0.

Since it's not directly related to the defaults handling in this PR I'll do it in a separate PR.

**Comment by etracer65** [comment]
It's going to be a bit of a clusterf*** any way you look at it. Because we **don't** want to set `gyro_lowpass_hz = 0` if the dynamic lpf is switch enabled, but we **do** if both the dynamic and static switches are disabled.

**Comment by McGiverGim** [comment]
Welcome to my world! 😋
I tried to make it easy. And for the user is more evident in CLI that the filter is deactivated.

**Pull request merged**